src/Model/Casts/CastResolver.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Mongolid\Model\Casts;
+
+use Mongolid\Model\Casts\DateTime\DateTimeCast;
+use Mongolid\Model\Casts\DateTime\ImmutableDateTimeCast;
+use Mongolid\Model\Casts\Exceptions\InvalidCastException;
+
+class CastResolver
+{
+    private const DATE_TIME = 'datetime';
+    private const IMMUTABLE_DATE_TIME = 'immutable_datetime';
+
+    public static array $validCasts = [
+        self::DATE_TIME,
+        self::IMMUTABLE_DATE_TIME,
+    ];
+
+    public static function resolve(?string $cast): ?CastInterface
+    {
+        if (is_null($cast)) {
+            return null;
+        }
```

Comment:
I think that if you can't accept null, you shouldn't even give this possibility in the parameter

Reply:
só para referência, discussão similar: https://github.com/leroy-merlin-br/mongolid/pull/220#discussion_r1364480671